### PR TITLE
Add events for claim metadata management operations

### PR DESCRIPTION
**Pull request by piraveena**
### Proposed changes in this pull request

Resolves https://github.com/wso2/product-is/issues/11705

### CODE REVIEW

File: components/claim-mgt/org.wso2.carbon.identity.claim.metadata.mgt/pom.xml [hunk 1]
```diff
@@ -132,6 +136,7 @@
                             !org.wso2.carbon.identity.claim.metadata.mgt.internal,
                             org.wso2.carbon.identity.claim.metadata.mgt.*;version="${carbon.identity.package.export.version}"
                         </Export-Package>
+                        <DynamicImport-Package>*</DynamicImport-Package>
```
Comment (emswbandara) [inline components/claim-mgt/org.wso2.carbon.identity.claim.metadata.mgt/pom.xml:139]: what is the reason for using DynamicImport-Package? can't we add the required imports under import-package section?

File: components/identity-event/org.wso2.carbon.identity.event/src/main/java/org/wso2/carbon/identity/event/internal/IdentityEventServiceComponent.java [hunk 2]
```diff
@@ -29,7 +29,7 @@
 import org.wso2.carbon.identity.event.handler.AbstractEventHandler;
 import org.wso2.carbon.identity.event.services.IdentityEventService;
 import org.wso2.carbon.identity.event.services.IdentityEventServiceImpl;
-import org.wso2.carbon.idp.mgt.IdpManager;
+//import org.wso2.carbon.idp.mgt.IdpManager;
```
Comment (emswbandara) [inline components/identity-event/org.wso2.carbon.identity.event/src/main/java/org/wso2/carbon/identity/event/internal/IdentityEventServiceComponent.java:32]: remove commented lines if not required

Comment (piraveena) [reply components/identity-event/org.wso2.carbon.identity.event/src/main/java/org/wso2/carbon/identity/event/internal/IdentityEventServiceComponent.java]: fixed with https://github.com/wso2/carbon-identity-framework/pull/3518/commits/65c22a6cceddbd6bde83d814e81c53535405398d

**Review by JKAUSHALYA — APPROVED** [review]

**Review by emswbandara — APPROVED** [review]

File: components/identity-event/org.wso2.carbon.identity.event/src/main/java/org/wso2/carbon/identity/event/internal/IdentityEventServiceComponent.java [hunk 3]
```diff
@@ -148,19 +147,5 @@ protected void setIdentityCoreInitializedEventService(IdentityCoreInitializedEve
     /* reference IdentityCoreInitializedEvent service to guarantee that this component will wait until identity core
          is started */
     }
-
```
Comment (IsuraD) [inline components/identity-event/org.wso2.carbon.identity.event/src/main/java/org/wso2/carbon/identity/event/internal/IdentityEventServiceComponent.java:151]: why did we remove this? 

This was to ,

 service to guarantee that this component will wait until identity core is started

**Review by IsuraD — APPROVED** [review]

File: components/identity-event/org.wso2.carbon.identity.event/src/main/java/org/wso2/carbon/identity/event/internal/IdentityEventServiceComponent.java (hunk 3 above)
Comment (piraveena) [reply components/identity-event/org.wso2.carbon.identity.event/src/main/java/org/wso2/carbon/identity/event/internal/IdentityEventServiceComponent.java:151]: That is not removed. remove idp.mgt componet which is not used and it cause cyclic dependency

**Pull request merged**